explorer/client/src/pages/transaction-result/TransactionResult.tsx

```diff
@@ -1,35 +1,100 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useLocation, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+    type SingleTransactionKind,
+    type Transfer,
+    type CertifiedTransaction,
+    isCertifiedTransaction,
+    isMoveCall,
+    isMoveModulePublish,
+    isSingleTransactionKind,
+    isTransfer,
+} from 'sui.js';
 
 import ErrorResult from '../../components/error-result/ErrorResult';
 import Longtext from '../../components/longtext/Longtext';
 import theme from '../../styles/theme.module.css';
+import { DefaultRpcClient as rpc } from '../../utils/api/DefaultRpcClient';
+import { type Loadable } from '../../utils/loadState';
 import { findDataFromID } from '../../utils/static/searchUtil';
 
 import styles from './TransactionResult.module.css';
 
-type DataType = {
-    id: string;
-    status: 'success' | 'fail' | 'pending';
-    sender: string;
-    created?: string[];
-    deleted?: string[];
-    mutated?: string[];
-    recipients: string[];
+const initState: Loadable<CertifiedTransaction> = {
+    loadState: 'pending',
+    transaction: {
+        data: {
+            kind: {
+                Single: {
+                    Transfer: {
+                        recipient: '',
+                        object_ref: ['', 0, ''],
+                    },
+                },
+            },
+            sender: '',
+            gas_payment: ['', 0, ''],
+            gas_budget: 0,
+        },
+        tx_signature: '',
+        auth_signature: '',
+    },
+    signatures: [],
 };
 
-function instanceOfDataType(object: any): object is DataType {
-    return (
-        object !== undefined &&
-        ['id', 'status', 'sender'].every((x) => x in object)
-    );
-}
+const isStatic = process.env.REACT_APP_DATA !== 'static';
 
 function TransactionResult() {
```

Comment:
This function is way too long and there are more logic to be added here, can we make it shorter by having a few more smaller functions/components?